### ISSUE TITLE
CartoonGAN Notes

### ISSUE THREAD

**Issue by GarlandZhang** [GarlandZhang/pg-toons#5]
http://openaccess.thecvf.com/content_cvpr_2018/papers/Chen_CartoonGAN_Generative_Adversarial_CVPR_2018_paper.pdf

## Introduction

"the power of Generative Adversarial Networks (GANs) formulated in a cyclic manner is explored to achieve high-quality style transfer, with the distinct feature that the model is trained using unpaired photos and stylized images"

"state-of-the-art methods fail to produce cartoonized images with acceptable quality. There are two reasons. First, instead of adding textures such as brush strokes in many other styles, cartoon images are highly simplified and abstracted from real-world photos. Second, despite variation of styles among artists, cartoon images have noticeable common appearance — clear edges, smooth color shading and relatively simple textures — which is very different from other forms of artworks"

**main contributions**: 

1. "We propose a dedicated GAN-based approach that effectively learns the mapping from real-world photos to cartoon images using unpaired image sets for training. [..] When cartoon images from individual artists are used for training, our method is able to reproduce their styles."

2. "We propose two simple yet effective loss functions in GAN-based architecture. In the generative network, to cope with substantial style variation between photos and cartoons, we introduce a semantic loss defined as an ℓ1 sparse regularization in the high-level feature maps of the VGG network [30]. In the discriminator network, we propose an edge-promoting adversarial loss for preserving clear edges"

## Related work

"Several works [5, 14, 16] have provided GAN solutions to pixel-to-pixel image synthesis problems. However, these methods require paired image sets for the training process which is impractical for stylization due to the challenge of obtaining such corresponding image sets."

"Given a style image and a content image, Gatys et al. [6] first proposed a neural style transfer (NST) method based on CNNs that transfers the style from the style image to the content image. They use the feature maps of a pre-trained VGG network to represent the content and optimize the result image, such that it retains the content from the content image while matching the texture information of the style image, where the texture is described using the global Gram matrix". The VGG network is a canvas to add texture detail from the style image while keeping the original image. However, we represent the texture using a Gram matrix.

**CycleGAN**:

"To address this fundamental limitation, CycleGAN [38] was recently proposed, which is a framework able to perform image translation with unpaired training data. To achieve this goal, it trains two sets of GAN models at the same time, mapping from class A to class B and from class B to class A, respectively. The loss is formulated based on the combined mapping that maps images to the same class. However, simultaneously training two GAN models often converges slowly, resulting in a time-consuming training process"

**CartoonGAN**:

Consists of 2 CNNS (1 for Generator, 1 for Discriminator). 

![model arch.](https://i.gyazo.com/a6ee13690f5a6ce00e3266a6dde3d20c.png)

"In CartoonGAN, the generator network G is used to map input images to the cartoon manifold.
Cartoon stylization is produced once the model is trained. G begins with a flat convolution stage followed by two down-convolution blocks to spatially compress and encode the images. Useful local signals are extracted in this stage for downstream transformation. Afterwards, eight residual blocks with identical layout are used to construct the content and manifold feature. We employ the residual block layout proposed in [15]. Finally, the output cartoon style images
are reconstructed by two up-convolution blocks which contain fractionally strided convolutional layer with stride 1/2 and a final convolutional layer with 7 × 7 kernels."

"Complementary to the generator network, the discriminator network D is used to judge whether the input image is a real cartoon image. Since judging whether an image is cartoon or not is a less demanding task, instead of a regular full-image discriminator, we use a simple patch-level discriminator with fewer parameters in D. Different from object classification, cartoon style discrimination relies on local features of the image. Accordingly, the network D is designed to be shallow. After the stage with flat layers, the network employs two strided convolutional blocks to reduce the resolution and encode essential local features for classification. Afterwards, a feature construction block and a 3 × 3 convolutional layer are used to obtain the classification response. Leaky ReLU (LReLU) [23] with α = 0.2 is used after each normalization layer"

**Loss Function**:

"The loss function L(G, D) in Eq.(1) consists of two parts: (1) the adversarial loss Ladv(G, D) (Section 3.2.1), which drives the generator network to achieve the desired manifold transformation, and (2) the content loss Lcon(G, D) (Section 3.2.2), which preserves the image content during cartoon stylization. We use a simple additive form for the loss function:

L(G, D) = Ladv(G, D) + ωLcon(G, D), where ω is the weight to balance the two given losses.
Larger ω leads to more content information from the input photos to be retained, and therefore, results in stylized images with more detailed textures [...] we set ω = 10 which achieves a good balance of style and content preservation"

**Ladv(G, D)**:

"applied to both networks G and D [...]  Its value indicates to what extent the output image of the generator G looks like a cartoon image"

"the task of the discriminator D is to figure out whether the input image is synthesized from the generator or from the real target manifold. However, we observe that simply training the discriminator D to separate generated and true cartoon images is not sufficient for transforming photos to cartoons. This is because the presentation of clear edges is an important characteristic of cartoon images, but the proportion of these edges is usually very small in the whole image. Therefore, an output image without clearly reproduced edges but with correct shading is likely to confuse the discriminator trained with a standard loss". Therefore, CartoonGan Discriminator is trained on edge-smoothed cartoons to not be confused with lack of edges.

**Lcon(G, D)**

"In addition to transformation between correct manifolds, one more important goal in cartoon stylization is to ensure the resulting cartoon images retain semantic content of the input photos". Basically, keep the general detail. 

**Initialization Phase**

"Note that the target of the generator network G is to reconstruct the input photo in a cartoon style while keeping the semantic content. We start the adversarial learning framework with a generator which only reconstructs the content of input images"

## Evaluation

"CartoonGAN is able to produce high-quality cartoon stylization using the data of individual artists for training, which are easily obtained from cartoon videos, since our method does not require paired images"
